src/containers/quiz/quiz-dialog/QuizDialog.tsx

```diff
@@ -0,0 +1,72 @@
+import { useTranslation } from 'react-i18next'
+
+import Box from '@mui/material/Box'
+import DialogTitle from '@mui/material/DialogTitle'
+import DialogContent from '@mui/material/DialogContent'
+import DialogActions from '@mui/material/DialogActions'
+import Dialog from '@mui/material/Dialog'
+import CloseRounded from '@mui/icons-material/CloseRounded'
+
+import { IconButton } from '~/design-system/components/icon-button/IconButton'
+import Button from '~/design-system/components/button/Button'
+
+import styles from '~/containers/quiz/quiz-dialog/QuizDialog.styles'
+
+export type QuizDialogProps = {
+  open: boolean
+  icon: React.ReactNode
+  title: string
+  description: string
+  actionText: string
+  descriptionParams?: Record<string, string>
+  secondaryActionText?: string
+  onAction: () => void
+  onSecondaryAction?: () => void
+  onClose?: () => void
+}
+
+const QuizDialog = ({
+  open,
+  icon,
+  title,
+  description,
+  actionText,
+  descriptionParams,
+  secondaryActionText,
+  onAction,
+  onSecondaryAction,
+  onClose
+}: QuizDialogProps) => {
```

Comment:
Change it to React.FC to be consistent with our codebase
```suggestion
}) => {
```